fe/fe-core/src/main/java/com/starrocks/sql/optimizer/transformer/SqlToScalarOperatorTranslator.java

```diff
@@ -213,7 +215,11 @@ private static class Visitor extends AstVisitor<ScalarOperator, Context> {
         private final CTETransformerContext cteContext;
         public final OptExprBuilder builder;
         public final Map<ScalarOperator, SubqueryOperator> subqueryPlaceholders;
-
+        // ArrayDeque will push into and pop out the head
+        // Example: the Deque is [1,2,3,4]
+        // when push(5) -> [5,1,2,3,4]
+        // then pop() -> [1,2,3,4]
+        private final Deque<Integer> usedSubFieldPos = new ArrayDeque<>();
```

Comment:
the theory is when the return value of `Expr` is the sub field of input value, this will work. but once the return value is new created complex type, `usedSubFieldPos` should be cleared

Reply:
we can define if a `Expr` which accepts former `usedSubFieldPos`. the definition is, when the return value of `Expr` is the subfield of input value, it accepts former `usedSubFieldPos`, otherwise, it don't.

Reply:
maybe default behavior should be not accept, so it won't return wrong result.